dessia_common/exports.py

```diff
@@ -253,3 +256,100 @@ def autosize_sheet_columns(sheet, min_width=5, max_width=30):
+
+
+class MarkdownWriter:
+    def __init__(self, print_limit: int = 25, table_limit: int = 12):
+        self.print_limit = print_limit
+        self.table_limit = table_limit
+
+    def _object_titles(self):
+        return ['Attribute', 'Type', 'Value'] #, 'Subvalues']
+
+    def _sequence_to_str(self, value: List[Union[list, dict, set]]):
+        in_values = value
+        if len(value) == 0:
+            return f"empty {type(value).__name__}"
+
+        printed_string = f"{len(value)} "
+        if isinstance(value, dict):
+            in_values = list(value.values())
+
+        all_class_names = list(set(subvalue.__class__.__name__ for subvalue in in_values))
+        if len(all_class_names) == 1:
+            printed_string += ''.join([f"{all_class_names[0]}",
+                                       f"{'s' if len(value) > 1 else ''}"])
+        else:
+            str_all_class = str(all_class_names).translate(str(all_class_names).maketrans('', '', "{}'"))
+            printed_string += f"{str_all_class}"
+
+        return printed_string
+
+    def _object_to_str(self, value) -> str:
+        if hasattr(value, 'name' ):
+            return f"{value.name}"
+        return 'unnamed'
+
+    def _value_to_str(self, value: Any) -> str:
+        if isinstance(value, (float, int, bool, complex)):
+            return str(round(value, 6))
+        if isinstance(value, str):
+            return (value if value != '' else 'no_value')
+        if isinstance(value, (list, dict, set)):
+            return self._sequence_to_str(value)
+        return self._object_to_str(value)
+
+    def _string_in_table(self, string: str = ''):
+        return string[:self.print_limit] + ('...' if len(string) > self.print_limit else '')
+
+    def _object_matrix(self, object_):
+        matrix = []
+        for attr, value in object_.__dict__.items():
+            matrix.append([attr,
+                           type(value).__name__,
+                           # (self._sequence_to_str(value) if isinstance(value, (list, dict, set)) else ' - '),
+                           (self._value_to_str(value) if not isinstance(value, (list, dict, set)) else ' - ')])
+        return matrix
+
+    def _head_table(self, col_names: List[str] = None) -> str:
+        cap_names = map(lambda x: x.capitalize(), col_names)
+        return ("| " + " | ".join(cap_names) + " |\n" +
+                "| ------ " * len(col_names) + "|\n")
+
+    def _table_line(self, row: List[Any]) -> str:
+        line = "|"
+        for value in row:
+            line += ''.join([" ", self._string_in_table(self._value_to_str(value)), " |"])
+        return line + "\n"
+
+    def _content_table(self, content: List[List[Any]]) -> str:
+        half_table = int(self.table_limit / 2)
+        table = ''
+        for row in content[:half_table]:
+            table += self._table_line(row)
+
+        if len(content) > self.table_limit and self.table_limit != -1:
```

Comment:
-1 stands for no limit ? Why not None in this case ?

Reply:
-1 allows me not to test if the value is None when i divide it by 2

Reply:
This seems made up to me. It looks like you are tricking your way out of the "table_limit is -1" without exposing the true concept of the code here. If I understand correctly, by taking the -1 example, you are doing this : 
```
self.table_limit = -1
half_table = int(self.table_limite / 2) = int(-0.5) = 0
table = ""
content[:half_table] = content[:0] = []
for row in []: # is thus uneffective and table is not built

[...]

if len(content) > self.table_limit and self.table_limit != -1:
   # table limit is -1 so we donnot go in here

[...]

# The same goes for the second part with 
for row in content[-half_table:]:
    # We don't loop here because content[-half_table:] is []

return table # Which is ""
```

Is that the expected result ? If yes, I would totally expose the true concept of the table_limit not being defined, here, which is : "If table limit is None, then we do nothing and show an empty string", simply by doing : 
```
table = ""
if self.table_limit is None:
    return table
# Then,  all the other stuff as you already did : 
half_table = int(self.table_limit / 2)
if ...
```

And I think this is even more true if the case for no table_limit is not to be an empty string, but to be the whole formatted dataset. In this case you don't even need the half table stuff, so this is another algorithm

Reply:
Changed the method following our oral conversation